### PR TITLE
blend module work in progress (WIP)

### PR DESCRIPTION
**Pull request by jywarren**
Basic blend module to combine previous step with the step before it -- mainly for use in conjunction with #255 so an imported image and the previous step can be blended together using a supplied function.

This is incomplete! Need to add a way to grab image 1 pixels from image 2 `PixelManipulation` call.

### CODE REVIEW

**Comment by jywarren** [comment]
For more on blend modes, see: https://en.wikipedia.org/wiki/Blend_modes

**Comment by jywarren** [comment]
@tech4GT @ccpandhare  👍

**Comment by jywarren** [comment]
Taking a look here, you can see why some of the step API functions are needed -- here we need to access the output from 2 steps prior, but there's not an easy way to find prior steps, or the array position of the current step. Trying to work around this with a closure! If you have ideas, please try building on this. 👍

**Comment by jywarren** [comment]
Basically at the moment we need to be able to `getStep(offset)` where offset of `-1` means the previous step, `-2` means the step before that, etc. Then we'll be able to fetch the output of any step with `getStep(-2).output` to use in the blending function.

**Comment by tech4GT** [comment]
@jywarren yeah, that makes sense!! I'll start working on it and submit a pr tomorrow maybe. Also @jywarren we need to keep a brainstorming session to discuss implementations and you can also fill me in on the refactored code.

**Comment by tech4GT** [comment]
@jywarren I am trying to build the .getStep() on the master branch, that's fine right?

**Comment by tech4GT** [comment]
@jywarren I want to suggest an alternate approach here, I think we can modify the input parameter we pass to the draw function of the module. This input can contain an array of all the previous step outputs and we can have a local variable input which we set to inputParameter.pop()
What do you say?

File: src/modules/Blend/Module.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    UI.onDraw(options.step);
+    var step = this;
+
+    // convert to runnable code:
+    if (typeof options.func === "string") eval('options.func = ' + options.func);
+
+    var getPixels = require('get-pixels');
+   
+    // save first image's pixels
+    var priorStep = util.getStep(-1);
```
Comment (jywarren) [inline src/modules/Blend/Module.js]: @tech4GT awesome -- next, can you use the `getStep()` API now available from #259 to fix this line, and remove the changes I'd attempted to implement `getStep()` in `AddStep.js`? Then let's do the big code tidying!

Comment (tech4GT) [reply src/modules/Blend/Module.js]: @jywarren yeah I'll rebase these changes over that pr

Comment (tech4GT) [reply src/modules/Blend/Module.js]: @jywarren Can I do that later today....actually I have a practical examination in 6 hours?

**Comment by jywarren** [comment]
i'm so sorry i didn't realize you were still in exams! No problem at all -
can wait until after. Good luck!

**Comment by tech4GT** [comment]
@jywarren Yeah just a couple of Lab examinations left but I can handle them, I'll do these changes later today and open a new PR maybe?

**Comment by tech4GT** [comment]
@jywarren this can be closed now

**Pull request closed without merging**